frontend/spec/features/agents_spec.rb

```diff
@@ -0,0 +1,1728 @@
+        element.fill_in with: "Agent Name #{now}"
+
+        within '#agent_person_contact_details' do
+          click_on 'Add Contact'
+        end
+
+        fill_in 'agent_agent_contacts__0__name_', with: "Contact Name #{now}"
+        fill_in 'agent_agent_contacts__0__email_', with: "Email Address #{now}"
+
+        # Click on save
+        find('button', text: 'Save Person', match: :first).click
+        element = find('.alert.alert-success.with-hide-alert')
+        expect(element.text).to eq "Agent Created"
+
+        element = find('h2')
+        expect(element.text).to eq "Agent Name #{now} Agent"
+      end
+
+      it 'can add multiple telephone numbers' do
+        now = Time.now.to_i
+
+        click_on 'Create'
+        click_on 'Agent'
+        click_on 'Person'
+
+        element = find('#agent_names__0__primary_name_')
+        element.fill_in with: "Agent Name #{now}"
+
+        within '#agent_person_contact_details' do
+          click_on 'Add Contact'
+        end
+
+        fill_in 'agent_agent_contacts__0__name_', with: "Contact Name #{now}"
+        fill_in 'agent_agent_contacts__0__email_', with: "Email Address #{now}"
+
+        within '#agent_agent_contacts__0__telephones_' do
+          click_on 'Add Telephone Number'
+          click_on 'Add Telephone Number'
+        end
+        fill_in 'agent_agent_contacts__0__telephones__0__number_', with: "Telephone Number 1 #{now}"
+        fill_in 'agent_agent_contacts__0__telephones__0__ext_', with: "Telephone Extension 1 #{now}"
+        fill_in 'agent_agent_contacts__0__telephones__1__number_', with: "Telephone Number 2 #{now}"
+        fill_in 'agent_agent_contacts__0__telephones__1__ext_', with: "Telephone Extension 2 #{now}"
+
+        # Click on save
+        find('button', text: 'Save Person', match: :first).click
+        element = find('.alert.alert-success.with-hide-alert')
+        expect(element.text).to eq "Agent Created"
+
+        expect(find('#agent_agent_contacts__0__telephones__0__number_').value).to eq "Telephone Number 1 #{now}"
+        expect(find('#agent_agent_contacts__0__telephones__0__ext_').value).to eq "Telephone Extension 1 #{now}"
+        expect(find('#agent_agent_contacts__0__telephones__1__number_').value).to eq "Telephone Number 2 #{now}"
+        expect(find('#agent_agent_contacts__0__telephones__1__ext_').value).to eq "Telephone Extension 2 #{now}"
+      end
+
+      it 'reports errors when updating a Person Agent with invalid data' do
+        agent = create(:agent_person)
+        visit "agents/agent_person/#{agent.id}/edit"
+
+        fill_in 'agent_names__0__primary_name_', with: ''
+
+        # Click on save
+        find('button', text: 'Save Person', match: :first).click
+        element = find('.alert.alert-danger.with-hide-alert')
+        expect(element.text).to eq 'Primary Part of Name - Property is required but was missing'
+      end
+
+      it 'reports errors when updating a Person Agent with invalid data' do
```

Comment:
This spec should be labeled 'can add a related agent', not the copied-from-above 'reports errors...'.